### DISCUSSION TITLE
What are the use cases for the new special registers?

### DISCUSSION THREAD

**Discussion by David-Else** [helix-editor/helix#7808] [General]
Currently on the master: https://docs.helix-editor.com/master/usage.html?highlight=register#registers

| Register character | When read              | When written             |
| ---                | ---                    | ---                      |
| `_`                | No values are returned | All values are discarded |
| `#`                | Selection indices (first selection is `1`, second is `2`, etc.) | This register is not writable |
| `.`                | Contents of the current selections | This register is not writable |
| `%`                | Name of the current file | This register is not writable |
| `*`                | Reads from the system clipboard | Joins and yanks to the system clipboard |
| `+`                | Reads from the primary clipboard | Joins and yanks to the primary clipboard |

But what actual use are these? For example, can we use the `%` register to send the name of the file out to a shell script? When would `#` ever be used, and how?

Cheers!

**Comment by pascalkuthe** [comment]
Current file registers is very useful for shell scripts or just in general using the filename. You can insert a register in a prompt so for example you can do `:sh <c-r>%` to execute the current file. Some people have been asking for a preselct option in the file picker (of the current file) that cab now be achieved by doing `<c-r>%` in the file picker. You can also copy the current election into the lront quickly now `<c-r>.`. These are just examles.

The black home register was around forever (its not new) and useful for using `d`/`c` without overwriting your current reigster.

The clipboard registers are jus the space as `<space>p`. We may want to remove the special commands for that in the future (so `<space>p` would be bound to the equivalent of `"*p` like in vim (internally that's already how they work now). You can now use the clipboard everywhere you can use a normal register (without us adding a billion new commands).

`#` is useful for counting. For example let's say you want to create a numbered list. You can select the content then di `a-s#Pi. `. This is especially nice when combined with the I cement/decrement commands (something similar was possible with them with some hacks in the last but now its much more streamlined).
